tests/conftest.py

```diff
@@ -95,8 +105,24 @@ def _import_and_add_jax(xp_available_backends: dict[str, types.ModuleType]) -> N
     msg = f"unsupported array backend: {GLASS_ARRAY_BACKEND}"
     raise ValueError(msg)
 
-# use this as a decorator for tests involving array API compatible functions
-array_api_compatible = pytest.mark.parametrize("xp", xp_available_backends.values())
+
+@pytest.fixture(
+    params=xp_available_backends.values(), ids=xp_available_backends.values()
+)
+def xp(request: pytest.FixtureRequest) -> types.ModuleType:
```

Comment:
This closes #611 too. I think the issue are so intertwined with each other that it is much easier to resolve both of them in a single PR.

Reply:
I think it would be nice to have docstrings for these fixtures. So at a glance I know what each one is doing.